State/ApplicationState.cpp

```diff
@@ -125,4 +130,76 @@ moonlight_xbox_dx::ApplicationState^ __stateInstance;
 moonlight_xbox_dx::ApplicationState^ moonlight_xbox_dx::GetApplicationState() {
 	if (__stateInstance == nullptr)__stateInstance = ref new moonlight_xbox_dx::ApplicationState();
 	return __stateInstance;
+}
+
+void moonlight_xbox_dx::ApplicationState::WakeHost(MoonlightHost^ host)
+{
+	if (host == nullptr || host->Connected) return;
+
+	if (host->MacAddress == nullptr || host->MacAddress == "00:00:00:00:00:00") {
+		Utils::Log("No recorded Mac address, the client and host must be connected at least once.\n");
+		return;
+	}
+
+	WSADATA wsa;
+	if (WSAStartup(MAKEWORD(2, 2), &wsa) != 0) {
+		Utils::Log("Websocket startup failed.\n");
+		return;
+	}
+
+	SOCKET s = socket(AF_INET, SOCK_DGRAM, IPPROTO_UDP);
+	if (s == SOCKET_ERROR || s == INVALID_SOCKET) {
+		Utils::Log("Websocket creation failed.\n");
+		WSACleanup();
+		return;
+	}
+	
+	std::string hostIP = Utils::PlatformStringToStdString(host->LastHostname).substr(0, 14);
+	std::string broadcastIP = (hostIP.substr(0, hostIP.length() - 3) + "255");
```

Comment:
This logic for handling the broadcast IP is wrong. I think you might want to check https://github.com/moonlight-stream/moonlight-qt/blob/ebe270bec5820b956f4c511fd31097a0c1d1a7c3/app/backend/nvcomputer.cpp#L216 to handle it better

Reply:
You can also use the both the https://github.com/TheElixZammuto/moonlight-xbox/blob/d622fe15e9bf591a97a7c69db0bb7ee4f132df61/State/MDNSHandler.cpp and https://learn.microsoft.com/en-us/uwp/api/windows.networking.connectivity.ipinformation.prefixlength?view=winrt-22621#windows-networking-connectivity-ipinformation-prefixlength property to correctly get the local IP address and the broadcast IP address